File: backend/lib/L1_domain/entities/goals/goal_import.py
#  Copyright (c) 2022. Alexandr Moroz
from __future__ import annotations

from dataclasses import dataclass

from .goal import Goal


# TODO: наверное, это должно быть в репозитории импорта и отдельная схема просто. Такая сущность это не ок
# TODO: может, лучше со схемами работать? Тогда ещё подумать как их протаскивать в юзкейс Л1
@dataclass
class GoalImport(Goal):
    parent: GoalImport | None = None

File: frontend/lib/L1_domain/entities/goals/goal_import.dart
// Copyright (c) 2022. Alexandr Moroz

// TODO: можно было отнаследоваться от Smartable. Но там обязательно id...
//  Возможно,если разнести Importable и PKGetable как на бэке,то получится упростить немного

class GoalImport {
  GoalImport({
    required this.code,
    required this.title,
    required this.description,
    required this.dueDate,
    required this.closed,
    this.selected = true,
  });

  final String code;
  final String title;
  final String description;
  final DateTime? dueDate;
  final bool closed;
  final bool selected;

  GoalImport copyWithSelected(bool _selected) => GoalImport(
        code: code,
        title: title,
        description: description,
        dueDate: dueDate,
        closed: closed,
        selected: _selected,
      );
}
